dbms/src/Storages/StorageKafka.cpp

```diff
@@ -89,7 +90,14 @@ class ReadBufferFromKafkaConsumer : public ReadBuffer
         // Consume message and mark the topic/partition offset
         // The offsets will be committed in the insertSuffix() method after the block is completed
         // If an exception is thrown before that would occur, the client will rejoin without comitting offsets
-        BufferBase::set(reinterpret_cast<char *>(msg->payload), msg->len, 0);
+        // BufferBase::set(reinterpret_cast<char *>(msg->payload), msg->len, 0);
+
+        // ReadBuffer will make messages into a stream without separate messages each other, 
+        // but for Row-based inputstream such as CSV or TSV, there should be a '\n' separator between each message;
+        msg_ctx = std::make_shared<String>(reinterpret_cast<char *>(msg->payload), msg->len);
+        if (msg_ctx->back() != '\n')
+            msg_ctx->append("\n");
```

Comment:
This won't work for RowBinary (and any non-text) format.

Why we have messages without terminating line feed in Kafka?
Is it possible that messages have line feeds in some use cases?
Is it possible to have many rows in single Kafka message?